src/main/java/com/intershop/oms/test/businessobject/order/OMSOrder.java

```diff
@@ -53,6 +59,54 @@ public class OMSOrder extends OMSBusinessObject implements OMSPropertyGroupOwner
     protected List<OMSShippingBucket> shippingBuckets = new ArrayList<>();
     protected OMSShop shop;
 
+    public OMSOrder shop(OMSShop shop)
+    {
+        setShop(shop);
+        return this;
+    }
+
+    public void setShop(OMSShop shop)
+    {
+        this.shop = shop;
+
+        Map<String, OMSTax> shopTaxes = shop.getShopTaxes();
+
+        if (sales != null)
+        {
+            sales.getUsedTaxes().forEach(t -> verifyPersistentTax(shopTaxes, t));
+        }
+        if (shippingBuckets != null)
+        {
+            shippingBuckets.forEach(sb -> sb.getUsedTaxes().forEach(t -> verifyPersistentTax(shopTaxes, t)));
+        }
+    }
+
+    public void verifyPersistentTax(Map<String, OMSTax> shopTaxes, OMSTax tax)
+    {
+        // any tax without a rate is expected as valid - the user should know, what he's doing
+        if (tax.getRate() == null)
```

Comment:
The problem is that the expected tax may be wrong if there is as shop mismatch.
e.g
	net=10
	gross=10.7
	=> expected tax= 7.0

if you have an order for a location with a tax of 7.5 (e.g. FullTax / USA / 7.7)
than following tax for the above price would be invalid:

(USA)
 "taxes" : [ {
          "type" : "FullTax",
          "amount" : 0.7,--instead of 7.5 for USA
        }
        
        
a complete test should hence compare the expected tax with the configured one, when the rate is NOT provided
so a method, like validateTax(String Location(*), BigDecimal expectedTax, String taxType)
Location: I'l l add a global comment  to the PR.